### DISCUSSION TITLE
Support Redis unix socket

### DISCUSSION THREAD

**Discussion by kommserv** [netbox-community/netbox#8757] [Ideas]
Presently, NetBox is limited to communicating with Redis TCP port, with no support for Redis unix socket.
This limitation feels unnecessary, and for someone familiar with `settings.py` it should be fairly easy to add support for socket.

For a quick-and-dirty hack, I can force NetBox to use socket *instead* of port by changing [this line](https://github.com/netbox-community/netbox/blob/06cb7f35f1698f6a0b43014e8bebe37a97237308/netbox/netbox/settings.py#L243):
```
'LOCATION': f'{CACHING_REDIS_PROTO}://{CACHING_REDIS_HOST}:{CACHING_REDIS_PORT}/{CACHING_REDIS_DATABASE}',
```

to
```
'LOCATION': f'unix://@{CACHING_REDIS_HOST}?db={CACHING_REDIS_DATABASE}',
```

This works, but is of course horrible as it breaks existing support for TCP. (The syntax was easily gleaned from comments in the other Python scripts).

A much neater approach has been suggested before, namely in [issue #4377](https://github.com/netbox-community/netbox/issues/4377), which seems to have been closed without any action taken. I think the outlined syntax in `configuration.py` is good, but the suggested changes to `settings.py` have of course become outdated.

And so I hope this post might inspire you, dear maintainer/developer, to bring Redis socket support to NetBox! :-)

**Comment by minijackson** [comment]
Hello, I'm also interested in supporting Redis through Unix sockets, and I'm willing to provide a patch. We [already have one](https://github.com/NixOS/nixpkgs/blob/dd1290b0f857782a60b251f89651c831cd3eef9d/pkgs/servers/web-apps/netbox/config.patch) in the Nixpkgs distribution, which has worked without any issues.

This defines a new `REDIS['tasks']['URL']` and `REDIS['caching']['URL']` which take priority if defined by the user. The user can then set this config if they want to use Unix sockets:

```python
REDIS = {
    "tasks": {"URL": "unix:///run/redis-netbox/redis.sock?db=0", "SSL": False},
    "caching": {"URL": "unix:///run/redis-netbox/redis.sock?db=1", "SSL": False},
}
```

What do you all think? Would this be accepted?

**Comment by jeremystretch** [comment]
@minijackson could you comment with your proposed implementation on #15962? That would be the best place to discuss.

**Reply by minijackson** [reply]
Done! Thanks for pointing me there
